fiftyone/core/dataset.py

```diff
@@ -3076,11 +3076,13 @@ def save_view(
         if view._root_dataset is not self:
             raise ValueError("Cannot save view into a different dataset")
 
-        url_name = self._validate_view_name(name, overwrite=overwrite)
+        view._set_name(name)
+        url_name = self._validate_saved_view_name(name, overwrite=overwrite)
 
         now = datetime.utcnow()
 
         view_doc = foo.ViewDocument(
+            id=ObjectId(),
```

Comment:
@brimoor I added an id field to the view document so that the front end has a unique identifier for saved_view components